File: wsl/useful_codes/c_vs_cpython_comparison/find_primes.c
#include <stdio.h>

int findPrimes(int num1, int num2) {
  printf("The Prime Numbers in (%d,%d) are:\n", num1, num2);

  for (int i = num1 + 1; i < num2; ++i) {
    int isPrime = 1;

    for (int j = 2; j <= i / 2; ++j) {
      if (i % j == 0) {
        isPrime = 0;
        break;
      }
    }

    if (isPrime == 1) {
      printf("%d\n", i);
    }
  }

  return 0;
}

int main(int argc, char *argv[]) {
  findPrimes(0, 50000);
  return 1;
}

File: wsl/useful_codes/c_vs_cpython_comparison/find_primes.py
def findPrimes(num1, num2):
    print("The Prime Numbers in ({}, {}) are:".format(num1, num2))

    for i in range(num1 + 1, num2):
        isPrime = True

        for j in range(2, i // 2 + 1):
            if i % j == 0:
                isPrime = False
                break

        if isPrime:
            print(i)

findPrimes(0, 1000000)
